proxy/sources/reddit.py

```diff
@@ -0,0 +1,157 @@
+            return None
+
+        api_data = json.loads(resp.text)
+        try: 
+            api_data = api_data["data"]["children"][0]["data"]
+
+            if not api_data["is_gallery"] or api_data["removed_by_category"] != None:
+                return None
+
+            try:
+                date = datetime.utcfromtimestamp(api_data["created"])
+            except ValueError:
+                date = datetime.now()
+
+            images = []
+            for image in api_data["gallery_data"]["items"]:
+                metadata = api_data["media_metadata"][image["media_id"]]
+                if metadata["status"] != "valid" or metadata["e"] != "Image":
+                    continue
+                url = self.image_url_handler(metadata["s"]["u"])
+                images.append(url)
+            if len(images) == 0:
+                return None
+
+        except KeyError:
+            return None
+
+        return {
+            "slug": meta_id,
+            "title": api_data["title"],
+            "description": "No description",
+            "author": "Unknown",
+            "artist": "Unknown",
+            "cover": images[0],
+            "groups": {"1": "Reddit"},
+            "chapter_dict": {
+                "1": {
+                    "volume": "1",
+                    "title": api_data["title"],
+                    "groups": { "1": images },
+                }
+            },
+            "chapter_list": [
+                [
+                    "1",
+                    "1",
+                    api_data["title"],
+                    "1",
+                    "No group",
+                    [
+                        date.year,
+                        date.month - 1,
+                        date.day,
+                        date.hour,
+                        date.minute,
+                        date.second,
+                    ],
+                    "1",
+                ],
+            ],
+            "pages_list": images,
+            "original_url": f"https://www.reddit.com{api_data['permalink']}",
+        }
+
+    @api_cache(prefix="reddit_api_dt", time=300)
+    def reddit_common(self, meta_id):
+        #return self.reddit_scrape(meta_id)
+        return self.reddit_api(meta_id)
+
+    @api_cache(prefix="reddit_series_dt", time=300)
+    def series_api_handler(self, meta_id):
+        data = self.reddit_common(meta_id)
+        return (
+            SeriesAPI(
+                slug=data["slug"],
+                title=data["title"],
+                description=data["description"],
+                author=data["author"],
+                artist=data["artist"],
+                groups=data["groups"],
+                cover=data["cover"],
+                chapters=data["chapter_dict"],
+            )
+            if data
+            else None
+        )
+
+    @api_cache(prefix="reddit_pages_dt", time=300)
+    def chapter_api_handler(self, meta_id):
+        data = self.imgur_common(meta_id)
```

Comment:
Should probably change this to call `self.reddit_common(meta_id)`. This one's super easy to miss though since you're returning the page list in the series API so this would only be called if it's directly invoked.